### PR TITLE
ASM-3012 - added support for Linux VM post OS processor

### PR DESCRIPTION
**Pull request by ksw2599**
modified puppet network module to take integer as the network resource title; the integer represents the sequence of the VM network adapter

modified the static.pp to ignore integer name/title

### CODE REVIEW

File: lib/puppet/parser/functions/get_seq_interface.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module Puppet::Parser::Functions
+  newfunction(:get_seq_interface, :type => :rvalue) do |args|
+    seq = args[0]
+    interfaces = lookupvar("interfaces")
+
+    interfaces.split(",").reject { |ifn| ifn == "lo" }[seq]
```
Comment (gavin-scott) [inline lib/puppet/parser/functions/get_seq_interface.rb]: Is the order of interfaces from `lookup("interfaces")` guaranteed? I would think you would want a sort here like `interfaces.split(",").reject { |ifn| ifn == "lo" }.sort[seq]`

Comment (ksw2599) [inline lib/puppet/parser/functions/get_seq_interface.rb]: So far, they seem to be ordered, but I can sort them if you would like me to.

Comment (gavin-scott) [inline lib/puppet/parser/functions/get_seq_interface.rb]: Yeah, that's probably not guaranteed. I would sort them.

File: lib/puppet/parser/functions/get_seq_interface.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  end
+end
```
Comment (gavin-scott) [inline lib/puppet/parser/functions/get_seq_interface.rb]: add a trailing new-line at end of file

File: manifests/init.pp [hunk 3]
```diff
@@ -139,7 +139,9 @@
 
   include '::network'
 
-  if is_mac_address($name){
+  if(type($name) == "integer") {
+    $interface = get_seq_interface($name)
+  } elsif is_mac_address($name){
```
Comment (gavin-scott) [inline manifests/init.pp]: while you're updating the PR I would fix the spacing on line 142 (`if (type`) and this line (`is_mac_address($name) {`.

**Comment by ksw2599** [comment]
@gavin-scott: They are fixed and amended.

**Comment by aimers1975** [comment]
@gavin-scott  Let me know if you want to wrap up this review, or if you want me to assign to someone else.

**Comment by gavin-scott** [comment]
+1 I don't think anything will break if this is merged before https://github.com/dell-asm/asm-deployer/pull/1204 , so going ahead and merging.

**Pull request merged**